library/pool.c

```diff
@@ -1184,46 +1197,129 @@ int dnet_io_init(struct dnet_node *n, struct dnet_config *cfg)
 	return err;
 }
 
+int dnet_server_backend_init(struct dnet_node *n, size_t backend_id)
```

Comment:
There is virtually ZERO performance issues on selecting element from tree or array of several elements. Even if you go crazy and create thousands of backends, this tiny difference is void compared to other costs.

But having a map of backends allows to add/remove them without reallocation and ugly loops with reinsertions/reallocations.